modules/python-modules/syslogng/modules/s3/s3_object.py

```diff
@@ -161,6 +165,8 @@ def __init__(
                 "compress",
                 "compresslevel",
                 "chunk-size",
+                "server-side-encryption",
+                "kms-key",
```

Comment:
I think this will cause existing JSON files to load.

e.g. if the JSON file was saved by a previous version of the S3 destination and we add this code, then the old file will not load, complaining about these missing fields.

These should be made optional, I think.

Reply:
Thanks. If I understand this correctly I only had to remove those entries from the `field` variable. Please correct me if I am wrong.

Reply:
You are right, and I can see you removed that check, with that this should be ready to be merged.

Reply:
If we are loading the JSON and do not have the `"kms-key"` field in it, `self.__kms_key` will be set to `None` by `self.__kms_key: str = cache.get("kms-key", kms_key)`, where we expect it to be a `str`. 

I think we should have an "upgrade" step, where we give defaults (`""`) to the JSON.

Something like:
```py
        if path:
            for upgrade_field in {
                "canned-acl",
                "server-side-encryption",
                "kms-key",
            }:
                cache[upgrade_field] = cache.get(upgrade_field, "")

            for field in {
                "persist-name",
                "bucket",
                "target-key",
                "timestamp",
                "index",
                "compress",
                "compresslevel",
                "chunk-size",
                "storage-class",
                "finished",
                "upload-id",
                "uploaded-parts",
                "pending-parts",
                "canned-acl",
                "server-side-encryption",
                "kms-key",
            }:
                try:
                    cache[field]
                except KeyError as e:
                    raise PersistLoadError from e
```

Did not test it though.

Reply:
This `__init__` function became really cryptic because of the 2 ctor paths, sorry for that.